feat(repo/changelogs): allow user configuration of source url (#16873)
renovatebot/renovate: docs/usage/configuration-options.md

}
```

> ### overwriteSourceUrl

Reviewer: Would it make things too complex internally if we simply used `sourceUrl`?

Author: Thats what I did first, but it is hard to maintain as I recall it just gets overwritten somewhere which I had no control of/in some other flow. so i just decided to have a new flag instead.

its a bit fuzzy as it was quite long time ago, but if needed i can try it again if needed.

Use this field to set the source URL for release notes retrieval.
Most commonly used for dependencies missing one.

Using this field we can specify the exact url to fetch release notes from.

Example setting source URL for package "dummy":

```json
{
  "packageRules": [
    {
      "matchPackageNames": ["dummy"],
      "overwriteSourceUrl": "https://github.com/org/dummy"
    }
  ]
}
```

<!-- prettier-ignore -->
!!! note

> Supported platforms are GitHub and GitLab only.

Reviewer: You mean that we support changelogs on github/gitlab only, right? i.e. someone using platform=azure could still configure this for dependencies which are hosted on github.com

Author: correct, 

![image](https://user-images.githubusercontent.com/97394622/184083481-32bbb98e-1c4d-4c0c-b1d9-2bfb645bdfec.png)

and we overwrite it before above block

}
```

### customChangelogUrl

> Use this field to set the source URL for release notes retrieval.

Reviewer: ```suggestion
Use this field to set the source URL for a package, including overriding an existing one.
Source URLs are necessary in order to look up release notes.
```

> Most commonly used for dependencies missing one.

Reviewer: ```suggestion
```

Using this field we can specify the exact url to fetch release notes from.

Example setting source URL for package "dummy":

```json
{
  "packageRules": [
    {
      "matchPackageNames": ["dummy"],

>       "overwriteSourceUrl": "https://github.com/org/dummy"

Reviewer: ```suggestion
      "customChangelogUrl": "https://github.com/org/dummy"
```

    }
  ]
}
```

<!-- prettier-ignore -->
!!! note

> Supported platforms are GitHub and GitLab only.

Reviewer: ```suggestion
Renovate can fetch changelogs from GitHub and GitLab platforms only, and setting the URL to an unsupported host/platform type won't change that.
```
